siobrultech_protocols/gem/protocol.py

```diff
@@ -15,16 +16,58 @@
 LOG = logging.getLogger(__name__)
 
 PACKET_HEADER = bytes.fromhex("feff")
+API_RESPONSE_WAIT_TIME_SECONDS = 3.0  # Time to wait for an API response
+PACKET_DELAY_CLEAR_TIME_SECONDS = 3.0  # Time to wait after a PDL request so that GEM can finish sending any pending packets
 
 
 class PacketProtocol(asyncio.Protocol):
-    def __init__(self, queue: asyncio.Queue):
+    def __init__(
+        self,
+        queue: asyncio.Queue,
+        on_connection_made: Optional[Callable[[GemApi], NoReturn]] = None,
```

Comment:
Today I learned about `NoReturn`!  Thanks :D

Reply:
Turns out it isn't actually what I wanted, since the callback is expected to return.